File: frontend/nordicwebhub-client/src/types/servicePackage.ts
export type ServicePackage = {
  id: number
  name: string
  description: string
  category: string
  monthlyPrice: number
  setupFee: number
  deliveryTime: string
  isActive: boolean
  createdAt: string
}

export type CreateServicePackageDto = {
  name: string
  description: string
  category: string
  monthlyPrice: number
  setupFee: number
  deliveryTime: string
  isActive: boolean
}

export type UpdateServicePackageDto = CreateServicePackageDto

File: backend/NordicWebHub.Api/Models/ServicePackage.cs
namespace NordicWebHub.Api.Models;

public class ServicePackage
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public decimal MonthlyPrice { get; set; }

    public decimal SetupFee { get; set; }

    public string DeliveryTime { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<ProjectRequest> ProjectRequests { get; set; } = new List<ProjectRequest>();

    public ICollection<ServiceOrder> ServiceOrders { get; set; } = new List<ServiceOrder>();
}
